### DISCUSSION TITLE
Can I manage workflows via git?

### DISCUSSION THREAD

**Discussion by minty99** [argoproj/argo-workflows#9866] [Q&A] [answered]
Can I manage workflows via git like ArgoCD? I want to save the workflow manifests in specific git repo and update the metadata automatically by git commit.

**Comment by terrytangyuan** [comment, marked as answer]
Use Argo Events to watch Git events and trigger workflows.

**Reply by minty99** [reply]
@terrytangyuan Can I "update" workflow manifests via Argo Events sensor? I found we can "trigger" an existing workflow, but nothing for "update".

**Reply by artemus717** [reply]
do you find a way to do this now ?

**Reply by agilgur5** [reply]
Workflows generally can't be updated once they're submitted (except for a few actions such as `suspend`). If you mean WorkflowTemplates, you can manage those via GitOps such as with Argo CD, as per #13147
